event.c

```diff
@@ -209,7 +210,39 @@ eq_debug_entry(const struct event_debug_entry *a,
 	return a->ptr == b->ptr;
 }
 
-int event_debug_mode_on_ = 0;
+#ifndef EVENT__DISABLE_DEBUG_MODE
+/*
+ * to avoid having to use an lock, encode the initialization status using
+ * an atomic variable.
+ */
+enum {
+	EVT_DBG_MODE_UNINITIALIZED = 0,
+	EVT_DBG_MODE_INITIALIZING,
+	EVT_DBG_MODE_INITIALIZED,
+	EVT_DBG_MODE_CLEANUP,
+};
+
+static atomic_init_ptr(void, event_debug_mode_state, NULL);
+
+int event_debug_mode_on(void)
+{
+	/*
+	 * event_debug_mode_too_late can be set from other threads while event_enable_debug_mode is running.
+	 * If the write gets in before the check, it will trigger - as expected.
+	 * If the write gets in late, these threads might slip through without learning about debug mode being enabled.
+	 * Luckily, state must be _INITIALIZING during and after the check so return true after waiting to make sure the resources are ready.
+	 *
+	 * This only affects impatient callers and after that the condition should never be met and play nice with branch prediction.
```

Comment:
i did run some tests but i have to admit that it can't be considered thorough.
my main concern was correct code (rather than speed).

this particular loop (and the functions depending on its result) should always go through the same.
as hinted at in the comment, the only way to branch off or run into the busy loop is by starting to work with events and only then calling `event_enable_debug_mode` - which isn't allowed anyway.

`atomic_load` is going to take longer than the plain read but it also ensures ordering when setting _too_late.